Sources/RTMP/RTMPStream.swift

```diff
@@ -489,31 +493,56 @@ open class RTMPStream: IOStream {
     }
 
     func close(withLockQueue: Bool) {
-        if withLockQueue {
-            lockQueue.sync {
-                self.close(withLockQueue: false)
+            if withLockQueue {
+                lockQueue.sync {
+                    self.close(withLockQueue: false)
+                }
+                return
             }
-            return
-        }
-        guard let connection, ReadyState.open.rawValue < readyState.rawValue else {
-            return
-        }
-        readyState = .open
-        if let fcPublishName {
-            connection.call("FCUnpublish", responder: nil, arguments: fcPublishName)
+            guard let connection, ReadyState.open.rawValue < readyState.rawValue else {
+                return
+            }
+            readyState = .open
+            if let fcPublishName {
+                connection.call("FCUnpublish", responder: nil, arguments: fcPublishName)
+            }
+            connection.doOutput(chunk: RTMPChunk(
+                                    type: .zero,
+                                    streamId: RTMPChunk.StreamID.command.rawValue,
+                                    message: RTMPCommandMessage(
+                                        streamId: 0,
+                                        transactionId: 0,
+                                        objectEncoding: self.objectEncoding,
+                                        commandName: "closeStream",
+                                        commandObject: nil,
+                                        arguments: [self.id]
+                                    )))
+        deleteStream()
```

Comment:
calling deleteStream after calling closeStream. I am unsure about the validity of this and have concerns about potential side effects. Could you tell me what tests you conducted

Reply:
We should separate close from deleteStream completely. I agree with @shogo4405 , there may be side effects of this.

Reply:
removed it deleteStream() from close function and add a separate function for it